common/txmgr/types/keystore.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+import "github.com/smartcontractkit/chainlink/core/services/pg"
+
+// KeyStore encompasses the subset of keystore used by txmgr
+type KeyStore[ADDR any, ID any, TX any, META any] interface {
+	CheckEnabled(address ADDR, chainID ID) error
+	GetNextMetadata(address ADDR, chainID ID, qopts ...pg.QOpt) (META, error)
```

Comment:
What is metadata?

Reply:
Some blockchains do not use an account nonce to prevent double-spending or replay attack (bitcoin, IOTA, Ripple for example don't).

Reply:
I think Sequence is a better general term than Metadata.
Sequence = nonce in evm and cosmos, UTXO in Bitcoin, etc.

Reply:
When you say "do not use an account nonce", do you mean they don't use an EVM-style nonce that is serially incremented?
Do they fit this more general definition?
> In [cryptography](https://en.wikipedia.org/wiki/Cryptography), a nonce is an arbitrary number that can be used just once in a cryptographic communication.

https://en.wikipedia.org/wiki/Cryptographic_nonce

It may still be worthwhile to avoid overloading the term - just trying to understand the model.